util/types/json/serdes.go

```diff
@@ -371,32 +345,12 @@ var jsonTypeCodeLength = map[byte]int{
 	typeCodeArray:   -1,
 	typeCodeLiteral: 1,
 	typeCodeInt64:   8,
-	typeCodeDouble:  8,
+	typeCodeFloat64: 8,
 	typeCodeString:  -1,
 }
 
-func jsonFromTypeCode(typeCode byte) JSON {
-	switch typeCode {
-	case typeCodeObject:
-		return new(jsonObject)
-	case typeCodeArray:
-		return new(jsonArray)
-	case typeCodeLiteral:
-		return new(jsonLiteral)
-	case typeCodeInt64:
-		return new(jsonInt64)
-	case typeCodeDouble:
-		return new(jsonDouble)
-	case typeCodeString:
-		return new(jsonString)
-	}
-	panic("unknown type code")
-}
-
-// Two map are equal if they have same keys and same values.
-// So we sort the keys before serialize in order to keep
-// their binary representations are same.
-func (m jsonObject) getSortedKeys() []string {
+// getSortedKeys returns sorted keys of a map.
+func getSortedKeys(m map[string]JSON) []string {
 	keys := make([]string, 0, len(m))
 	for k := range m {
```

Comment:
The order of  this operation is undetermined, call `getSortedKeys` multiple times you may get different result!